File: C/Partie6/Exo26a.c
#include <stdio.h>
#include <stdlib.h>
#include <fcntl.h>
#include <unistd.h>

int main()
{
    int n;
    printf("Nombre d'entiers : ");
    scanf("%d", &n);
    int *tab = malloc(n * sizeof(int));
    printf("Entrez les %d entiers :\n", n);
    for (int i = 0; i < n; i++)
    {
        scanf("%d", &tab[i]);
    }
    char nom[100];
    printf("Nom du fichier de sauvegarde : ");
    scanf("%s", nom);
    int fd = open(nom, O_WRONLY | O_CREAT | O_TRUNC, 0624);
    if (fd == -1)
    {
        perror("open");
        free(tab);
        exit(1);
    }
    write(fd, tab, n * sizeof(int));
    close(fd);
    free(tab);
    return 0;
}

File: Python/Partie6/Exo26a.py
import struct
import os

n = int(input("Nombre d'entiers : "))
tab = []
for i in range(n):
    tab.append(int(input(f"Entier {i+1} : ")))
nom = input("Nom du fichier de sauvegarde : ")
with open(nom, "wb") as f:
    for val in tab:
        f.write(struct.pack('i', val))

os.chmod(nom, 0o664)